packages/astro/src/core/preview/vite-plugin-astro-preview.ts

```diff
@@ -24,7 +24,8 @@ export function vitePluginAstroPreview(settings: AstroSettings): Plugin {
 					return;
 				}
 
-				const pathname = stripBase(req.url!, base);
+				const strippedPathname = stripBase(req.url!, base);
+        const pathname = new URL(strippedPathname, 'https://a.b').pathname
```

Comment:
I'm not familiar with the codebase but what is this `a.b` doing?

Reply:
It's anything, we don't care about the domain. We just extract the pathname but need the URL method to correctly parse the pathname.